### PR TITLE
Dave branch

### PR DESCRIPTION
**Pull request by saie100**
## Description

[Describe the changes that you've made in this pull request and explain why they are necessary.]

## Checklist

Please ensure that your pull request meets the following requirements:

- [ ] Code compiles without errors and warnings
- [ ] Code follows the project's coding style guidelines
- [ ] Code is tested and test cases pass
- [ ] Code has been reviewed by at least one other developer

## Related Issues

[If your pull request is related to any issues, link them here.]

## Additional Notes

[Include any additional information or notes that may be helpful in reviewing your pull request.]

### CODE REVIEW

File: src/Executive.cpp [hunk 1]
```diff
@@ -13,43 +13,105 @@ using std::endl;
...
+  this->tempDir = "../test_data/temp_data/"; // default temp directory
+  this->outputDir = "../test_data/output_data/"; // default output directory
+  this->reduceDLL = "";
+  this->mapDLL = "";
   
-  // exit program if there are not 3 arguments
-  if (count == 1) {
-    cerr << "Error: No arguments entered, need 3\nHelp: main [input dir] [temp dir] [output dir]" << endl;
-    exit(1);
+  // loop through arguements and set arguements equal to
```
Comment (mcockle23) [inline src/Executive.cpp:27]: nit: arguments

File: src/FileManager.cpp [hunk 2]
```diff
@@ -223,3 +223,18 @@ vector<string> FileManager::getFilesFromDir(string dirPath) {
     exit(1);
   }
 }
+
+// deltes all text files within a directory
```
Comment (mcockle23) [inline src/FileManager.cpp:227]: nit: deletes

File: src/headers/Executive.h [hunk 3]
```diff
@@ -21,20 +21,31 @@ class Executive {
...
+  string getReduceDLL();
+
+  // returns mapDLL member variable
+  string getMapDLL();
 
  private:
-  /***************************** Members **************************************/
-  int argumentCount; // stores the number of arguments
-  vector<string> arguments; // stores a list of arguments
+  /***************************** Members Variables **************************************/
```
Comment (mcockle23) [inline src/headers/Executive.h:40]: nit: Member

File: src/headers/FileManager.h [hunk 4]
```diff
@@ -48,6 +48,10 @@ class FileManager {
   // parameter dirPath is the path to the directory
   static vector<string> getFilesFromDir(string dirPath);
 
+  // deltes all text files within a directory
```
Comment (mcockle23) [inline src/headers/FileManager.h:51]: nit: deletes

File: tests/ExecutiveTest.cpp [hunk 5]
```diff
@@ -19,55 +19,50 @@ TEST(ExecutiveTest, NoArgument){
...
+    command[3] = kwarg2.data();
+    command[4] = arg2.data();
+    
     char ** pointer = command;
     
-    EXPECT_DEATH({
-        Executive commandline(arguementCount +1, pointer); 
-        commandline.getArguments();
-        }, "Error: 2 arguments entered, need 3");
+    EXPECT_DEATH(Executive commandline(arguementCount*2 +1, pointer), "--dllDir keyword argument required");
```
Comment (mcockle23) [inline tests/ExecutiveTest.cpp:65]: you'll need to update this to be the mapdll or reduce dll instead of dlldir

Comment (mcockle23) [reply tests/ExecutiveTest.cpp:65]: Also add a test for the other one

File: tests/FileManagerTest.cpp [hunk 6]
```diff
@@ -45,6 +45,49 @@ TEST(FileManagerTest, ReadFromFile){
...
+    FileManager::writeFile(FileManager::CREATE, "./temp_test/file3.txt", "hello world 3");
+    
+    // verify files were created
+    vector<string> list = {"./temp_test/file1.txt","./temp_test/file2.txt","./temp_test/file3.txt"};
+    EXPECT_EQ(FileManager::getFilesFromDir("./temp_test/"), list);
+    
+    // delete all files from directory
+    FileManager::deleteFilesFromDir("./temp_test/");
+    
+    // verify files were created
```
Comment (mcockle23) [inline tests/FileManagerTest.cpp:84]: nit: verify files were deleted

File: tests/FileManagerTest.cpp [hunk 7]
```diff
@@ -45,6 +45,49 @@ TEST(FileManagerTest, ReadFromFile){
...
+
+    // create a temporary directory for testing
+    system("mkdir temp_test");
+
+    // create three files
+    FileManager::writeFile(FileManager::CREATE, "./temp_test/file1.txt", "hello world 1");
+    FileManager::writeFile(FileManager::CREATE, "./temp_test/file2.txt", "hello world 2");
+    FileManager::writeFile(FileManager::CREATE, "./temp_test/file3.txt", "hello world 3");
+    
+    // verify files were created
```
Comment (mcockle23) [inline tests/FileManagerTest.cpp:58]: nit: verify files were deleted

**Review by mcockle23 — CHANGES_REQUESTED** [review]
Mostly looks good. Just a few comments to fix and add separate tests for the two dll arguments instead of the one dll argument you had originally

**Review by vnijat93 — APPROVED** [review]
lgtm

**Review by mcockle23 — APPROVED** [review]

**Pull request merged**